client/src/pages/files/addTags.tsx

```diff
@@ -0,0 +1,198 @@
+import Prose from 'components/Prose';
+import React, {
+  useState,
+  useEffect,
+  KeyboardEvent,
+  useRef,
+  useMemo,
+} from 'react';
+
+import {getUserFiles, updateDocumentTags} from 'lib/api/files';
+import {UserFile} from 'types/UserFile';
+import {Table} from 'semantic-ui-react';
+
+import {useAuth} from 'contexts/auth';
+
+export default function TaggingView() {
+  const {user} = useAuth();
+  const [userFiles, setUserFiles] = useState<Map<string, [UserFile, boolean]>>(
+    new Map<string, [UserFile, boolean]>()
+  );
+
+  useEffect(() => {
+    if (user) {
+      getUserUploadedFiles();
+    }
+  }, [user]);
+
+  async function getUserUploadedFiles() {
+    const newUserFiles = await getUserFiles(user!);
+    setUserFiles(
+      new Map(
+        Array.from(newUserFiles).map(userFile => {
+          return [userFile.fileName, [userFile, false]];
+        })
+      )
+    );
+  }
+
+  const removeTagFromFile = (tagIndex: number, fileName: string) => {
+    const newFiles = new Map(userFiles);
+    const oldFileInfo = newFiles.get(fileName)!;
+    const newFileInfo: UserFile = {
```

Comment:
Repeating a lot of information from the previous object, can do this instead:
```
const newFileInfo: UserFile = {
  ...oldFileInfo,
  tags: oldFileInfo.tags.slice()
}
```
Which copies all fields from the previous object and overwrites tags. (This is assuming you made the above change)